src/main/java/com/iexec/worker/replicate/ReplicateDemandService.java

```diff
@@ -81,21 +84,27 @@ public ReplicateDemandService(IexecHubService iexecHubService,
      * to execute the task.
      */
     void askForReplicate() {
-        log.debug("Asking for a new replicate");
-        // TODO check blocknumber only once a replicate is received.
-        long lastAvailableBlockNumber = iexecHubService.getLatestBlockNumber();
-        if (lastAvailableBlockNumber == 0) {
-            log.error("Cannot ask for new tasks, your blockchain node is not synchronized");
-            return;
+        if (askForReplicateLock.tryLock()) {
```

Comment:
How about this format ?
```
        if (!askForReplicateLock.tryLock()) {
            return;
        }
```
This way branches of the code are clearer with less nested if/else statements.

Reply:
Sure, good idea. Thanks! Fixed in https://github.com/iExecBlockchainComputing/iexec-worker/pull/433/commits/8e33cf27c66adce5ac54cac466b766650cd47fe1.